### ISSUE TITLE
Mintmenu does not honor xkb keybindings in Linux Mint MATE 19

### ISSUE THREAD

**Issue by Sl0vi** [linuxmint/mintmenu#196]
I have edited my keyboard layout, so that the menu button on my keyboard is rebound to be the Super_L button but the menu does not open when I press the menu button on my keyboard.

I remapped the menu button by editing the file /usr/share/X11/xkb/symbols/fo to look like the following:

    default  partial alphanumeric_keys
    xkb_symbols "basic" {
    
        include "latin(type2)"

        name[Group1]= "Faroese";
    
        key <TLDE>	{ [   onehalf,   section, threequarters,    paragraph ]	};
        key <LSGT>	{ [      less,    greater,    backslash,      notsign ]	};
        key <SPCE>	{ [     space,      space, nobreakspace, nobreakspace ] };

        key <AE05>	{ [         5,    percent,     NoSymbol,         cent ] };
        key <AE06>	{ [         6,  ampersand,          yen,     NoSymbol ] };
        key <AE11>	{ [      plus,   question,    plusminus, questiondown ] };
        key <AE12>	{ [dead_acute, dead_grave,          bar,    brokenbar ] };
    
        key <AB03>	{ [         c,          C,    copyright,     NoSymbol ] };
        key <AB10>	{ [     minus, underscore,       hyphen,       macron ] };
    
        key <AC03>	{ [         d,          D,          eth,          ETH ] };
        key	<AC10>	{ [        ae,	       AE	] };
        key <AC11>	{ [    oslash,   Ooblique 	] };
    
        key <AD08>	{ [         i,          I       ] };
        key <AD11>	{ [     aring,      Aring, dead_diaeresis, dead_circumflex ] };
        key <AD12>	{ [       eth,        ETH,   dead_tilde,   dead_caron ] };
    
        key <BKSL>	{ [apostrophe,   asterisk, dead_doubleacute, multiply ]	};
    
        key <MENU> { [ Super_L ] };
    
        include "level3(ralt_switch)"
    };
    
    partial alphanumeric_keys
        xkb_symbols "nodeadkeys" {
    
        include "fo(basic)"
    
        name[Group1]= "Faroese (eliminate dead keys)";
    
        key <AE12>	{ [     acute,      grave,          bar,       ogonek ]	};
        key <AD11>	{ [     aring,      Aring,    diaeresis,       degree ]	};
        key <AD12>	{ [ diaeresis, asciicircum,  asciitilde,       macron ]	};
        key <AC10>	{ [        ae,         AE,        acute,  doubleacute ]	};
        key <AC11>	{ [    oslash,   Ooblique,  asciicircum,        caron ]	};
        key <BKSL>	{ [apostrophe,   asterisk,  doubleacute,     multiply ]	};
        key <AB08>	{ [     comma,  semicolon,      cedilla,       ogonek ]	};
        key <AB09>	{ [    period,      colon, periodcentered,   abovedot ]	};
    };

The important part is line 30, it maps the MENU key to Super_L.

If I try to change the keyboard shortcut for the mintmenu in menu preferences it correctly sets the shortcut to Super_L if I press the menu keyboard button, but pressing the Menu button still doesn't open the mintmenu.

The showkey command shows that the keycode for Super_L is 125 while the keycode for Menu is 127. Has mintmenu been changed to read keycodes instead of button mappings?

I have recently upgraded to Linux Mint 19 MATE. This bug was not present in earlier versions of Linux Mint MATE.

**Comment by Sl0vi** [comment]
Ok, actually this bug is in older versions of Linux Mint. I have been using MATE+i3 with <Super>+Ctrl_L keybinding for over a year and have just switched back to just using MATE that I didn't realise that the bug was always there.

**Comment by clefebvre** [comment]
Yes, afaik it reads keycodes. The code for the keybinding is in https://github.com/linuxmint/mintmenu/blob/master/usr/lib/linuxmint/mintMenu/keybinding.py.

**Comment by Sl0vi** [comment]
Sorry for the slow reply, haven't been on github for a long time.

Are you suggesting that I fix the keycodes myself? Is this a bug or is it intended behaviour?

I'm not very good at python, but I can try and see if I can fix it.

**Comment by clefebvre** [comment]
Hi @Sl0vi, 

Yes if you can. It was written that way so I guess it's both a bug and the intended behaviour. It would be better if it didn't rely on codes, although it's not an important issues since it's a very corner case.
